tests/all_tests.sh

```diff
@@ -1,13 +1,9 @@
 #!/bin/bash
 
-srcdir="$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)"
+cd "$(dirname "$0")"/..
+jobs=${JOBS:-4}
+filter=${FILTER:-""}
 
-test_status=0
-test_files=( "$srcdir"/tests/tokenized/*.txt )
-for tf in "${test_files[@]}"; do
-    test_name="$(basename "$tf")"
-    test_name="${test_name%.txt}"
-    "$srcdir"/tests/decompyle_test.sh $test_name tests || test_status=1
-done
-
-exit $test_status
+find tests/tokenized -type f -name '*.txt' -a -name "*${filter}*" -print0 | \
+    xargs -0 -I '{}' -P $jobs \
+          bash -c 'echo -e "$(tests/decompyle_test.sh "$(basename -s .txt "{}")" tests-out)"'
```

Comment:
This no longer fails when the tests fail...  The exit status must be preserved and propagated (perhaps just a `set -e` is sufficient?)

Reply:
I also had to rework the inner subshell:
```diff
- bash -c 'echo -e "$(tests/decompyle_test.sh "$(basename -s .txt "{}")" tests-out)"'
+ bash -c 'o=$(tests/decompyle_test.sh "$(basename -s .txt "{}")" tests-out) r=$?; echo "$o"; exit $r'

```